### DISCUSSION TITLE
ORCID auth?

### DISCUSSION THREAD

**Discussion by zchandler** [usnistgov/NEMO#59] [Ideas]
Are there any plans on the roadmap for integrating with https://orcid.org/ for authentication? I reckon that would be very helpful in many communities. I am new to NEMO, sorry if this has already been addressed elsewhere.

**Comment by rptmat57** [comment]
Hi Zach,
As long as ORCID provides a verified email address, you should be able to use [NEMO-allauth](https://pypi.org/project/NEMO-allauth/) plugin to authenticate with it. The email will have to match the email in NEMO.

NEMO-allauth uses django-allauth behind the scenes, so take a look at https://django-allauth.readthedocs.io/en/latest/providers.html#orcid for specific ORCID settings that would need to be setup.

**Reply by zchandler** [reply]
Cool, thanks Mathieu. I am reading through https://django-allauth.readthedocs.io/en/latest/providers.html#orcid now. 
Since this is django-allauth documentation, I am guessing that for implementation the NEMO (client) side would have to implement ORCID as a service provider? If I am right about that, I would highly recommend this approach for the main NEMO project (guessing that other stakeholder might agree). Is this something that might get into the main branch, or would we have to add that to a fork?

ORCID is more of a comprehensive identity system (and more oriented to scientific research use cases), and I think generally relies on email addresses at home institutions. If NEMO were to leverage ORCID API, there might also be ways to use it for authorization (does this person have the right training to access this facility, etc.) so it might be worth looking into. Many of our researchers already use ORCID, so I just wanted to put it on your radar. https://info.orcid.org/documentation/

Thanks!

**Reply by rptmat57** [reply]
[NEMO-allauth](https://pypi.org/project/NEMO-allauth/) does that. It hooks up both NEMO and django-allauth. no need to fork, you can use NEMO as is, and follow the instructions to install NEMO-allauth.
if you are using docker, follow the instructions [here](https://github.com/usnistgov/NEMO/wiki/Plugin-list) to install the plugin without having to recreate the docker image.

I don't believe interacting with ORCID API is in the plans for NEMO at NIST, but that would make for a great plugin if somebody wanted to look into that!

**Reply by zchandler** [reply]
Thanks again Mathieu. Appreciate the guidance. Depending on what our development plans are this year, maybe we can look into that ORCID plugin! Thanks for laying out the opportunity.
